scripts/scripts.js

```diff
@@ -72,11 +72,28 @@ async function loadEager(doc) {
   }
 }
 
+/**
+ * sets the Content-Security-Policy meta tag to the document based on JSON file
+ */
+
+async function setCSP() {
+  const resp = await fetch(`${window.hlx.codeBasePath}/scripts/csp.json`);
+  const json = await resp.json();
+  const directives = Object.keys(json);
+  const policy = directives.map((directive) => `${directive} ${json[directive].join(' ')}`).join('; ');
+  const meta = document.createElement('meta');
+  meta.setAttribute('http-equiv', 'Content-Security-Policy');
+  meta.setAttribute('content', policy);
+  document.head.appendChild(meta);
```

Comment:
```suggestion
  document.addEventListener("securitypolicyviolation", (e) => sampleRUM("csperror", {source: `${e.documentURI}:${e.lineNumber}:${e.columnNumber}`, target: e.blockedURI }));
  document.head.appendChild(meta);
```

This would allow us to track CSP violations without overloading the endpoint as the same sampling applies as other RUM checkpoints.